src/SSHDebugPS/ProcFSOutputParser.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Windows.Documents;
+
+namespace Microsoft.SSHDebugPS
+{
+    public class ProcFSOutputParser
+    {
+        public static string EscapedCommandText => CommandText.Replace("$", "\\$");
+        public static string CommandText => @"echo shell-process:$$; for filename in /proc/[0-9]*; do echo $filename,cmdline:$(tr '\0' ' ' < $filename/cmdline 2>/dev/null),ls:$(ls -lh $filename/exe 2>/dev/null); done";
```

Comment:
>$filename [](start = 107, length = 9)

I am not sure if it is worth it, but shellcheck.net suggests quoting all these various commands/variables to avoid problems with filenames that have spaces...

Reply:
i think your regex covers that and there shouldn't be a space in `filename`